lib/active_record/virtual_attributes/virtual_delegates.rb

```diff
@@ -261,7 +261,9 @@ def self.select_from_alias(to_ref, col, to_model_col_name, src_model_id)
         yield arel if block_given?
 
         # convert arel to sql to populate with bind variables
-        ::Arel::Nodes::Grouping.new(Arel.sql(arel.to_sql))
+        conn = to_ref.klass.connection
```

Comment:
Is this evaluated at load time via `virtual_delegate :description, :to => :fancy_photo, :prefix => true, :type => :string` or lazy evaluated later?  If it's load time, is this hitting the database at require time?

Reply:
nope. lazy. every time